server/catgenome/src/main/java/com/epam/catgenome/util/MotifSearcher.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2021 EPAM Systems
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.epam.catgenome.util;
+
+import com.epam.catgenome.component.MessageHelper;
+import com.epam.catgenome.constant.MessagesConstants;
+import com.epam.catgenome.entity.reference.motif.Motif;
+
+import java.util.Spliterators;
+import java.util.Spliterator;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public final class MotifSearcher {
+
+    private static final int NEGATIVE_STRAND = 1;
+    private static final byte CAPITAL_A = 'A';
+    private static final byte CAPITAL_C = 'C';
+    private static final byte CAPITAL_G = 'G';
+    private static final byte CAPITAL_T = 'T';
+    private static final byte CAPITAL_N = 'N';
+    private static final byte LOWERCASE_A = 'a';
+    private static final byte LOWERCASE_C = 'c';
+    private static final byte LOWERCASE_G = 'g';
+    private static final byte LOWERCASE_T = 't';
+    private static final byte LOWERCASE_N = 'n';
+
+    private MotifSearcher() {}
+
+    public static List<Motif> search(final byte[] seq, final String regex, String contig) {
```

Comment:
lets add new parameter `strand` and if it is provided - will search only for such strand (it is needed to be able to provide for UI only strand specific results (in case of track view for example))